src/Avalonia.FreeDesktop/DBusMenuExporter.cs

```diff
@@ -248,17 +275,24 @@ object GetProperty((NativeMenuItemBase item, NativeMenu menu) i, string name)
                         if (item.ToggleType != NativeMenuItemToggleType.None)
                             return item.IsChecked ? 1 : 0;
                     }
-
+                    
                     if (name == "icon-data")
                     {
                         if (item.Icon != null)
                         {
-                            var ms = new MemoryStream();
-                            item.Icon.Save(ms);
-                            return ms.ToArray();
+                            var loader = AvaloniaLocator.Current.GetService<IPlatformIconLoader>();
```

Comment:
Does it still work with the global menu this way?

Reply:
yes... global menu also want working unless we used ico format.